src/lang/queryAst/getSafeInsertIndex.ts

```diff
@@ -0,0 +1,22 @@
+import { getIdentifiersInProgram } from './getIndentifiersInProgram'
+import { Program, Expr } from 'lang/wasm'
+import { Node } from '@rust/kcl-lib/bindings/Node'
+
+/**
+ * Given a target expression, return the body index of the last-used variable
+ * within the provided program.
+ */
+export function getSafeInsertIndex(
+  targetExpr: Node<Program | Expr>,
+  program: Node<Program>
+) {
+  const identifiers = getIdentifiersInProgram(targetExpr)
+  const insertIndex = identifiers.reduce((acc, curr) => {
+    const bodyIndex = program.body.findIndex(
+      (a) =>
+        a.type === 'VariableDeclaration' && a.declaration.id?.name === curr.name
+    )
+    return Math.max(acc, bodyIndex + 1)
+  }, 0)
+  return insertIndex
+}
```

Comment:
problem with `TagDeclarator`s is that they are not just on the body like variables.

I think maybe creating a `getTagDecloratorsInProgram` function that returns 
`{tagName: string, expressionIndex}[]`
so that you don't have to hunt for it's index in the body later.

Loose thoughts not dug into it.

Reply:
Yeah exactly I think it's gotta be a different utility, no worries I'll write some unit tests for it.